server/src/main/java/io/unitycatalog/server/service/credential/aws/AwsCredentialVendor.java

```diff
@@ -0,0 +1,28 @@
+package io.unitycatalog.server.service.credential.aws;
+
+import io.unitycatalog.server.exception.BaseException;
+import io.unitycatalog.server.exception.ErrorCode;
+import io.unitycatalog.server.persist.utils.ServerPropertiesUtils;
+import io.unitycatalog.server.service.credential.CredentialContext;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+
+import java.util.Map;
+
+public class AwsCredentialVendor {
+
+  private final Map<String, S3StorageConfig> s3Configurations;
+
+  public AwsCredentialVendor() {
+    this.s3Configurations = ServerPropertiesUtils.getInstance().getS3Configurations();
```

Comment:
nit: Do we want to support dynamically adding configurations to `servers.properties` w/o restarting? If so should just do this in `vendAwsCredentials()`

Reply:
yeah, maybe can do this in a follow-up PR, but also wonder if should just hold off on this since the configs _should_ be moving to database/entities/etc.

Reply:
> If so should just do this in vendAwsCredentials()

@creechy `ServerPropertiesUtils` loads the properties file when the static instance is initialized. So, moving this to `vendAwsCredentials` by itself won't support dynamically adding configs, no?